Tribler/Test/test_remote_search.py

```diff
@@ -1,150 +1,110 @@
+            if isinstance(community, AllChannelCommunity) and self.channel_search or \
+                isinstance(community, SearchCommunity) and self.torrent_search:
+                connections = community.get_nr_connections()
+                break
 
-            self.guiUtility.utility.write_config('showsaveas', 0)
+        if connections >= 10:
+            self.connections_lc.stop()
+            self.conn_deferred.callback(None)
 
-            self.frame.top_bg.OnDownload()
-            self.CallConditional(
-                120, lambda: self.frame.librarylist.GetNrResults() > 0, do_assert, 'no download in librarylist',
-                tribler_session=self.guiUtility.utility.session, dump_statistics=True)
 
-        def do_select():
-            self.assert_(self.frame.searchlist.GetNrResults() > 0, 'no hits matching ubuntu + desktop',
-                         tribler_session=self.guiUtility.utility.session, dump_statistics=True)
-            self.screenshot('After doing ubuntu search + desktop filter, got %d results' %
-                            self.frame.searchlist.GetNrResults())
-            items = self.frame.searchlist.GetItems()
-            keys = items.keys()
+class TestRemoteChannelSearch(BaseTestSearch):
+    """
+    This class contains tests to test remote channel search.
+    """
 
-            self.frame.searchlist.Select(keys[0])
-            self.callLater(5, do_download)
+    def setUp(self, autoload_discovery=True):
+        BaseTestSearch.setUp(self, autoload_discovery=autoload_discovery)
+        self.channel_search = True
 
-        def do_filter():
-            self.assert_(self.frame.searchlist.GetNrResults() > 0, 'no hits matching ubuntu + desktop',
-                         tribler_session=self.guiUtility.utility.session, dump_statistics=True)
-            self.screenshot('After doing ubuntu search, got %d results' % self.frame.searchlist.GetNrResults())
-            self.frame.searchlist.GotFilter('desktop')
+    @deferred(timeout=300)
+    def test_remote_channel_search(self):
+        """
+        Testing whether remote channels can be found when executing a search query in AllChannel community
+        """
+        def perform_search(_):
+            self.session.search_remote_channels([u'de'])
 
-            self.callLater(5, do_select)
+        self.conn_deferred.addCallback(perform_search)
+        return self.search_deferred
 
-        def do_search():
-            self.guiUtility.toggleFamilyFilter(newState=False, setCheck=True)
-            self.guiUtility.dosearch(u'ubuntu')
-            self.callLater(10, do_filter)
 
-        self.startTest(do_search)
+class TestRemoteTorrentSearch(BaseTestSearch):
+    """
+    This class contains tests to test remote torrent search.
+    """
+
+    def setUp(self, autoload_discovery=True):
+        BaseTestSearch.setUp(self, autoload_discovery=autoload_discovery)
+        self.torrent_search = True
+
+    @deferred(timeout=300)
```

Comment:
Unfortunately, this test takes a long time (around one minute) since it's starting up a Dispersy session and connection to other peers. This is why the timeout is that high.

We should refactor these tests so they are not dependent on the external Dispersy network but this is quite involving and probably too much work for the remaining time of my thesis.